File: Cpp/testPolynomials.cxx
#include <iostream>
#include <vector>

#include "polynomial.hh"

template <typename T>
void printVec(std::vector<T> v);

int main(void){

  std::cout << "Making the polynomial 2x^2 + 5x - 3" << std::endl;
  std::vector<double> coeffs= {2,5,-3};
  
  Polynomial *p = new Polynomial(coeffs);
  
  //Using all the print functions:
  std::cout << "using printOrder" << std::endl;
  p->printOrder();
  std::cout << "printing coeffs" << std::endl;
  p->printCoeffs();
  std::cout << "printing coeff [1]" << std::endl;
  p->printSingleCoeff(1);
  std::cout << "And the whole polynomial" << std::endl;
  p->printPolynomial();
  
  std::cout<< "Using the accessor functions" << std::endl;
  std::cout << "Polynomial p of order " << p->getOrder() << " has as 2nd element " << p->getSingleCoeff(1) << std::endl;
  printVec(p->getCoeffs());
 
  //using the copy constructor:
  Polynomial p2(*p);
  std::cout << "Copy of polynomial, calling printvec" << std::endl;
  p2.printPolynomial();

  
  return 0;
}
  

template <typename T>
void printVec(std::vector<T> v){
  std::cout << "Contents of Vector: ";
  for(auto x : v){
    std::cout << x << "  ";
  }
  std::cout << std::endl;
}

File: Java/src/TestPolynomials.java
import java.io.*;
import java.lang.Number;
import java.util.List;
import java.util.ArrayList;


public class TestPolynomials {

    public static void main(String[] args){
	double[] d1 = {1,2,3};
	ArrayList c1 = new ArrayList();
	for(double d : d1){
	    c1.add(new Double(d));
	}

	Polynomial p1 = new Polynomial(c1);
	
	System.out.println("Printing the full list of coefficients using the getCoeffs accessor");
	for(Double c : p1.getCoeffs()){
	    System.out.print(c.toString() + ", ");
	}
	System.out.println();
    
	System.out.println("Getting the 2nd coeff using the getSingleCoeff function: Coeff[1] = " + p1.getSingleCoeff(1).toString());
	System.out.println("Gettign the order using getOrder: Order is " + Integer.toString(p1.getOrder()));

	System.out.println("Now testing the various print functions:");
	System.out.println("printPolynomial results in:");
	p1.printPolynomial();
	System.out.println("printCoeffs results in:");
	p1.printCoeffs();
	System.out.println("printSingleCoeff(1) results in:");
	p1.printSingleCoeff(1);
	System.out.println("printOrder results in:");
	p1.printOrder();
    	
    }

}

	
